src/Tickets/Admin/Tickets/Page.php

```diff
@@ -233,26 +246,193 @@ public function get_url( array $args = [] ): string {
 	}
 
 	/**
-	 * Adds the Event Tickets All Tickets page.
+	 * Get the page title.
 	 *
-	 * @since 5.14.0
+	 * @since TBD
 	 */
-	public function add_tec_tickets_admin_tickets_page() {
-		$admin_pages = tribe( 'admin.pages' );
+	public function get_the_page_title(): string {
+		return __( 'All Tickets', 'event-tickets' );
+	}
+
+	/**
+	 * Get the menu title.
+	 *
+	 * @since TBD
+	 */
+	public function get_the_menu_title(): string {
+		return __( 'All Tickets', 'event-tickets' );
+	}
+
+	/**
+	 * Get the parent page slug.
+	 *
+	 * @since TBD
+	 */
+	public function get_parent_page_slug(): string {
+		return static::$parent_slug;
+	}
+
+	/**
+	 * Get the menu position.
+	 *
+	 * @since TBD
+	 */
+	public function get_position(): ?int {
+		return 1.2;
+	}
+
+	/**
+	 * Maybe register the tabs if we're on our page.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function maybe_register_tabs(): void {
+		if ( ! static::is_on_page() ) {
+			return;
+		}
+
+		$this->register_tabs();
+	}
+
+	/**
+	 * Register a tab for the page.
+	 * This method is public to allow external plugins to register their own tabs.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $slug  The tab's slug (used in URL and as key).
+	 * @param string $label The tab's label.
+	 * @param array  $args  {
+	 *     Optional. Array of tab arguments.
+	 *
+	 *     @type bool   $visible    Whether the tab should be visible. Default true.
+	 *     @type string $capability The capability required to see this tab. Default 'manage_options'.
+	 *     @type bool   $active     Whether this is the active tab. Default false.
+	 * }
+	 */
+	public function add_tab( string $slug, string $label, array $args = [] ): void {
+		$this->register_tab( $slug, $label, $args );
+	}
 
-		$admin_pages->register_page(
+	/**
+	 * Register the tabs for the page.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	protected function register_tabs(): void {
+		/**
+		 * Action that fires before registering the admin tabs.
+		 *
+		 * @since TBD
+		 *
+		 * @param Page $this The current page instance.
+		 */
+		do_action( 'tec_tickets_admin_before_register_tabs', $this );
+
+		$this->register_tab(
+			'ticket-table',
+			__( 'Event Tickets', 'event-tickets' ),
 			[
-				'id'       => static::$slug,
-				'path'     => static::$slug,
-				'parent'   => static::$parent_slug,
-				'title'    => esc_html__( 'All Tickets', 'event-tickets' ),
-				'position' => 1.2,
-				'callback' => [
-					$this,
-					'render_tec_tickets_admin_tickets_page',
-				],
+				'visible'    => true,
+				'capability' => $this->required_capability(),
+				'active'     => true,
 			]
 		);
+
+		/**
+		 * Action that fires after registering the admin tabs.
+		 * Use this hook to register additional tabs.
+		 *
+		 * @since TBD
+		 *
+		 * @param Page $this The current page instance.
+		 */
+		do_action( 'tec_tickets_admin_after_register_tabs', $this );
```

Comment:
```suggestion
		do_action( 'tec_tickets_admin_tickets_page_after_register_tabs', $this );
```

Assuming this is only impacting the `All Tickets` page.